http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -1520,44 +1520,54 @@ class LeoPubsubMessageSubscriber[F[_]](
           )
       }
 
-      updateApp = msg.cloudContext match {
+      updateApp = (msg.cloudContext match {
         case CloudContext.Gcp(_) =>
           getGkeAlgFromRegistry()
             .updateAndPollApp(UpdateAppParams(msg.appId, msg.appName, latestAppChartVersion, msg.googleProject))
         case CloudContext.Azure(azureContext) =>
           azurePubsubHandler
             .updateAndPollApp(msg.appId, msg.appName, latestAppChartVersion, msg.workspaceId, azureContext)
+      }).flatMap { _ =>
+        updateAppLogQuery
+          .update(msg.appId, msg.jobId, UpdateAppJobStatus.Success, Some(ctx.now))
+          .transaction
+          .void
       }
 
       updateAppWithErrorHandling = updateApp
-        .handleErrorWith {
-          // Fatal case (as in, the app is no longer usable), polling app liveness failed
-          // The two fatal cases are included separately, because later we may wish to fail fatally on `HelmException`, but roll back on `AppUpdatePollingException`
-          // This would provide more cases in which an app is left in a usable state
-          // Note that an app can also emit this error if the liveness probe fails before an update is triggered, so rolling back may not have an effect
-          case e: AppUpdatePollingException => F.raiseError(e)
-          // Fatal case, helm call failed for either listener or app charts
-          case e: HelmException => F.raiseError(e)
-          // Non fatal catch-all case, set app status back to running but append whatever error occurred in db for traceability
-          case e =>
-            for {
-              _ <- appQuery.updateStatus(msg.appId, AppStatus.Running).transaction
-              // You cannot update this log wording without updating the corresponding prod and staging alerts here
-              //     https://console.cloud.google.com/monitoring/alerting/policies/8184448493858086363?project=broad-dsde-prod
-              //     https://console.cloud.google.com/monitoring/alerting/policies/16136890211426349187?project=broad-dsde-staging
-              errorContext =
-                s"Error updating Azure app with id ${msg.appId.id} and cloudContext ${msg.cloudContext.asString}"
-              _ <- logger.warn(ctx.loggingCtx, e)(errorContext)
-              error = AppError(
-                s"${errorContext}: ${e.getMessage}",
-                ctx.now,
-                ErrorAction.UpdateApp,
-                ErrorSource.App,
-                None,
-                Some(ctx.traceId)
```

Comment:
It looks like this traceId would be the only way to connect this specific error to our app-update record; we have the app ID but I don't think we would be able to tell which errors went with which update events. Would it make sense to include the traceId on the app update log, or maybe the appErrorId?

Reply:
I guess we could use the timestamp but that feels a bit muddy.

Reply:
Hmm... I was originally planning to sort by timestamp, but I can see your point. I was trying to avoid duplication of including the entire message, but I like your idea of including the app error id.

Reply:
+1 to adding the app error id, it will make the join to the error table a lot easier